src/pages/masternodes.tsx

```diff
@@ -0,0 +1,87 @@
+import { ForwardedRef, forwardRef, PropsWithChildren, useCallback, useRef, useState } from 'react'
+import { masternodes } from '@defichain/whale-api-client'
+import { Loader } from '../components/shared/Loader'
+import { useMasternodes } from '../hooks/useMasternodes'
+
+interface MasternodeItemProps extends PropsWithChildren<unknown> {
+  data: masternodes.MasternodeData
+}
+
+const MasternodeItem = forwardRef(function masternodes ({ data }: MasternodeItemProps, ref: ForwardedRef<HTMLDivElement>) {
+  return (
+    <div
+      data-testid='masternode_item'
+      ref={ref}
+      className='flex flex-col md:flex-row justify-center md:justify-between border-b border-gray-300'
+    >
+      <div className='p-4 text-center md:text-left  md:w-1/4 break-all'>{data.owner.address}</div>
+      <div className='p-4 text-center md:text-right md:w-1/4 break-all'>{data.operator.address}</div>
+      <div className='p-4 text-center md:text-right'>{data.creation.height}</div>
+      <div className='p-4 text-center md:text-right'>{data.resign.height}</div>
+      <div className='p-4 text-center md:text-right'>{data.mintedBlocks}</div>
+      <div className='p-4 text-center md:text-right'>{data.state}</div>
+    </div>
+  )
+})
+
+export default function MasternodePage (): JSX.Element {
+  const [nextToken, setNextToken] = useState<string>('')
+  const observer = useRef<IntersectionObserver>()
+  const {
+    error,
+    hasNext,
+    loading,
+    masternodes,
+    next
+  } = useMasternodes(5, nextToken)
```

Comment:
https://nextjs.org/docs/api-reference/data-fetching/getInitialProps

Reply:
@fuxingloh , we can not use Nextjs data-fetching method because `useWhaleClient` is a hook and can only be used inside page components.